src/validators/SignedKeyRequestValidator.sol

```diff
@@ -0,0 +1,136 @@
+import {IdRegistry} from "../IdRegistry.sol";
+
+contract SignedKeyRequestValidator is IMetadataValidator, Ownable2Step, EIP712 {
+    /*//////////////////////////////////////////////////////////////
+                                 STRUCTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice Signed key request specific metadata.
+     *
+     *  @param requestingFid The fid of the entity requesting to add
+     *                       a signer key.
+     *  @param requestSigner Signer address. Must be the owner of
+     *                       the requestingFid fid.
+     *  @param signature     EIP-712 SignedKeyRequest signature.
+     */
+    struct SignedKeyRequest {
+        uint256 requestingFid;
+        address requestSigner;
+        bytes signature;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Emit an event when the admin sets a new IdRegistry contract address.
+     *
+     * @param oldIdRegistry The previous IdRegistry address.
+     * @param newIdRegistry The ne IdRegistry address.
+     */
+    event SetIdRegistry(address oldIdRegistry, address newIdRegistry);
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    bytes32 internal constant _METADATA_TYPEHASH =
+        keccak256("SignedKeyRequest(uint256 userFid,uint256 requestingFid,bytes signerPubKey)");
+
+    /*//////////////////////////////////////////////////////////////
+                                STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev The IdRegistry contract.
+     */
+    IdRegistryLike public idRegistry;
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the IdRegistry and owner.
+     *
+     * @param _idRegistry   IdRegistry contract address.
+     * @param _initialOwner Initial contract owner address.
+     */
+    constructor(address _idRegistry, address _initialOwner) EIP712("Farcaster MetadataValidator", "1") {
+        idRegistry = IdRegistryLike(_idRegistry);
+        _transferOwnership(_initialOwner);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                               VALIDATION
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Validate the SignedKeyRequest metadata associated with a signer key.
+     *         (Key type 1, Metadata type 1)
+     *
+     * @param userFid               The fid of the end user adding this signer key.
+     * @param signerPubKey          The public key of the signer.
```

Comment:
the name `signerPubKey` is slightly confusing since the terms signer and publicKey seem to mostly have been dropped in favor of key

Reply:
Yeah, this the EdDSA public key that the signer is adding to the key registry. The validator checks that the metadata field is a signed ECDSA message that recovers to the owner of `requestingFid`.

Reply:
The naming in here is pretty rough, since there's both the protocol-level signer and the metadata signature. If you have suggestions for better names in here, they are very welcome.

Reply:
Yea I think since we're in the context of the `SignedKeyRequest` the prefix `signer` may be more misleading than useful since it refers to an signer from an "outer scope".

Since we use `key` instead of `pubKey` in the `add` and `addFor` methods my suggestion is just `key` so the name stays consistent. But also I'm less familiar so this might have it's own collisions.

another option, though it feels heavy, `requestedFid, requestedKey, requestingFid`